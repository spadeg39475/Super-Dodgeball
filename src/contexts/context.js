import React, { useState, createContext } from 'react';


export const GameContext = createContext({
    game: null,
    setGame: ()=>{}
});

const GameContextProvider = (props) => {
    
    const setGame = (game)=>{
        setState({...state, game: game})
    }
    
    const initState = {
        game: null,
        setGame:setGame
    }
    
    const [state, setState ] = useState(initState)
    
    

    
     
    return (
        <GameContext.Provider value={{state, setGame:state.setGame}}>
            {props.children}
        </GameContext.Provider> 
    )

}

export default GameContextProvider;