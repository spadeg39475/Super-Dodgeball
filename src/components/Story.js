import React, { useState, useEffect, useContext} from "react";
import { GameContext } from '../contexts/context'
import Footer from './Footer';

import '../css/story.css'

const Story = () => {

    const {state, setGame} = useContext(GameContext);
    if(state.game){
        state.game.destroy(true,false)
        setGame(null);
    } 
    return(
        <div className='story' id='story'>
            <div className='block'>
                <div className='subtitle'>故事背景</div>
                <p className='text'>
                    <span className='l1'>
                        興趣是打架，運動萬能的不良少年國雄<br/>
                    </span>    
                    <span className='l2'>
                        在全國高校闘球試合上遇到了宿敵 － 花園高校<br/>
                    </span>
                    <span className='l3'>
                        目標是奪得全國優勝....
                    </span>
                </p>
            </div>
            
            <div className="img">
                <img src='./materials/img/story-pic.PNG'></img>
            </div>
        </div>
    )
}

export default Story;