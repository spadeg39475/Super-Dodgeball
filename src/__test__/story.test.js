import React from 'react';
import Story from '../components/Story';
import renderer from 'react-test-renderer';
import {GameContext} from '../contexts/context'




describe('about', ()=>{
    
    it('should show story component', ()=>{
        const tree = renderer.create(
            <GameContext.Provider value={{state: {game:null}}}>
                <Story/>
            </GameContext.Provider>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
})


