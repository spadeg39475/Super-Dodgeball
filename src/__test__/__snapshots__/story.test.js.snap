// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about should show story component 1`] = `
<div
  className="story"
  id="story"
>
  <div
    className="block"
  >
    <div
      className="subtitle"
    >
      故事背景
    </div>
    <p
      className="text"
    >
      <span
        className="l1"
      >
        興趣是打架，運動萬能的不良少年國雄
        <br />
      </span>
      <span
        className="l2"
      >
        在全國高校闘球試合上遇到了宿敵 － 花園高校
        <br />
      </span>
      <span
        className="l3"
      >
        目標是奪得全國優勝....
      </span>
    </p>
  </div>
  <div
    className="img"
  >
    <img
      src="./materials/img/story-pic.PNG"
    />
  </div>
</div>
`;
